core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -153,7 +156,7 @@ class SearchService(
   def refreshResolver(): Unit = resolver.update()
 
   def persist(check: FileCheck, symbols: List[FqnSymbol], commitIndex: Boolean, boost: Boolean): Future[Option[Int]] = {
-    val iwork = Future { blocking { index.persist(check, symbols, commitIndex, boost) } }
+    val iwork = Future { blocking { index.persist(check, symbols, commitIndex, boost) } }(ex)
```

Comment:
this should not be bounded. The thing we need to back pressure is the indexing (which generates the memory overhead), not the persisting (which clears the heap).